crates/sui-sdk/examples/utils.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_config::{
+    sui_config_dir, Config, PersistedConfig, SUI_CLIENT_CONFIG, SUI_KEYSTORE_FILENAME,
+};
+use sui_json_rpc_types::Coin;
+use sui_keys::keystore::{AccountKeystore, FileBasedKeystore};
+use sui_sdk::{
+    sui_client_config::{SuiClientConfig, SuiEnv},
+    wallet_context::WalletContext,
+};
+
+use futures::{future, stream::StreamExt};
+use tracing::info;
+
+use reqwest::Client;
+use serde_json::json;
+use shared_crypto::intent::Intent;
+use sui_sdk::types::{
+    base_types::SuiAddress,
+    crypto::SignatureScheme::ED25519,
+    digests::TransactionDigest,
+    programmable_transaction_builder::ProgrammableTransactionBuilder,
+    quorum_driver_types::ExecuteTransactionRequestType,
+    transaction::{Argument, Command, Transaction, TransactionData},
+};
+use sui_sdk::{rpc_types::SuiTransactionBlockResponseOptions, SuiClient, SuiClientBuilder};
+
+// const SUI_FAUCET: &str = "https://faucet.devnet.sui.io/gas"; // devnet faucet
+
+pub const SUI_FAUCET: &str = "https://faucet.testnet.sui.io/v1/gas"; // testnet faucet
+
+// if you use the sui-test-validator and use the local network; if it does not work, try with port 5003.
+// const SUI_FAUCET: &str = "http://127.0.0.1:9123/gas";
+
+/// Return a sui client to interact with the APIs,
+/// the active address, and another address that can be used as a recipient.
+pub async fn setup_for_write() -> Result<(SuiClient, SuiAddress, SuiAddress), anyhow::Error> {
+    let (client, active_address) = setup_for_read().await?;
+    let wallet = retrieve_wallet().await?;
+    let addresses = wallet.get_addresses();
+    let addresses = addresses
+        .into_iter()
+        .filter(|address| address != &active_address)
+        .collect::<Vec<_>>();
+    let recipient = addresses
+        .get(0)
+        .expect("Cannot get the recipient address needed for writing operations. Aborting");
+
+    Ok((client, active_address, *recipient))
+}
+
+/// Return a sui client to interact with the APIs, and the active address.
+pub async fn setup_for_read() -> Result<(SuiClient, SuiAddress), anyhow::Error> {
+    let client = SuiClientBuilder::default().build_testnet().await?;
+    println!("Sui testnet version is: {}", client.api_version());
+    let mut wallet = retrieve_wallet().await?;
+    assert!(wallet.get_addresses().len() >= 2);
+    let active_address = wallet.active_address()?;
+
+    // make sure we have some SUI (5_000_000 MIST) on this address
+    let coin = fetch_coin(&client, &active_address).await?;
+
+    if coin.is_none() {
+        request_tokens_from_faucet(active_address).await?;
+    }
```

Comment:
Good observation. Ideally, for the coin read api the address should have some coins, but the coin read api itself does not do any writes. My take was to make sure that there is at least a coin such that one can see some output. If you think it's better to have it in the `write` part, I'll move it.

Reply:
Ah okay -- that's reasonable, I can get behind that.  In that case, I would just document what both these setup functions are doing, because the `read` path needs a coin for a different reason than the `write` path.